test/unittests/execute_numeric_test.cpp

```diff
@@ -16,12 +16,11 @@ using namespace fizzy::test;
 
 namespace
 {
-TypedExecutionResult execute_unary_operation(Instr instr, TypedValue arg)
+auto create_unary_operation_executor(Instr instr)
 {
     const auto& instr_type = get_instruction_type_table()[static_cast<uint8_t>(instr)];
     EXPECT_EQ(instr_type.inputs.size(), 1);
     EXPECT_EQ(instr_type.outputs.size(), 1);
-    EXPECT_EQ(instr_type.inputs[0], arg.type);
```

Comment:
Should this check go inside lambda now?

Reply:
The `test::execute()` checks types inside.